src/h5web/toolbar/controls/DomainSlider/DataHistogram.tsx

```diff
@@ -1,43 +1,54 @@
 import { AxisBottom, AxisLeft } from '@visx/axis';
 import { scaleLinear } from '@visx/scale';
 import { useDomain } from '../../../vis-packs/core/hooks';
-import type { Size } from '../../../vis-packs/core/models';
+import type { ScaleType, Size, Domain } from '../../../vis-packs/core/models';
+import { H5WEB_SCALES } from '../../../vis-packs/core/scales';
 import styles from './DataHistogram.module.css';
 
 interface Props {
   bins: number[];
   data: number[];
   size: Size | undefined;
+  scaleType: ScaleType;
+  sliderDomain: Domain;
 }
 
 function DataHistogram(props: Props) {
-  const { data, bins, size = { width: 0, height: 0 } } = props;
+  const {
+    data,
+    bins,
+    scaleType,
+    sliderDomain,
+    size = { width: 0, height: 0 },
+  } = props;
   const { height, width } = size;
 
-  const domain = useDomain(data);
+  const histDomain = useDomain(data);
 
-  const xScale = scaleLinear({
-    domain: [bins[0], bins[bins.length - 1]],
+  const xScale = H5WEB_SCALES[scaleType].createScale({
+    domain: sliderDomain,
     range: [0, width],
   });
   const yScale = scaleLinear({
-    domain: domain ? [0, domain[1]] : undefined,
+    domain: histDomain ? [0, histDomain[1]] : undefined,
     range: [0, height],
   });
 
   return (
     <div className={styles.container}>
       <svg width={width} height={height} className={styles.svg}>
-        {data.map((d, i) => (
-          <rect
-            className={styles.bar}
-            key={i} // eslint-disable-line react/no-array-index-key
-            x={xScale(bins[i])}
-            y={height - yScale(d)}
-            width={xScale(bins[i + 1]) - xScale(bins[i])}
-            height={yScale(d)}
-          />
-        ))}
+        <svg width={width} height={height}>
+          {data.map((d, i) => (
+            <rect
+              className={styles.bar}
+              key={i} // eslint-disable-line react/no-array-index-key
+              x={xScale(bins[i])}
+              y={height - yScale(d)}
+              width={xScale(bins[i + 1]) - xScale(bins[i])}
+              height={yScale(d)}
+            />
+          ))}
+        </svg>
```

Comment:
Enclosing the bars in a new `svg` that has `overflow: hidden`. Otherwise, bars overflow from the parent svg that needs to have to `overflow: visible` for the axes.

Reply:
Can you use a group (`g`) instead of a nested `svg`?

Reply:
I don't think you can put CSS properties to a group ? Or that there is a SVG equivalent to `overflow: hidden` ?

Reply:
Boom! `<g clip-path="inset(0 0 0 0) view-box">` 😤